otel-worker-core/src/api/client.rs

```diff
@@ -163,16 +170,31 @@ impl ApiClient {
     ) -> Result<models::TraceSummary, ApiClientError<TraceGetError>> {
         let path = format!("v1/traces/{}", trace_id.as_ref());
 
-        self.do_req(Method::GET, path, api_result).await
+        self.do_req(Method::GET, path, None, api_result).await
     }
 
     /// List a summary traces
     pub async fn trace_list(
         &self,
+        limit: Option<u32>,
+        time: Option<time::OffsetDateTime>,
     ) -> Result<Vec<models::TraceSummary>, ApiClientError<TraceGetError>> {
         let path = "v1/traces";
 
-        self.do_req(Method::GET, path, api_result).await
+        let mut map = BTreeMap::new();
+
+        if let Some(limit) = limit {
+            map.insert("limit", limit.to_string());
+        }
+
+        if let Some(time) = time {
+            let formatted = time
+                .format(&Rfc3339)
+                .map_err(|_| ApiClientError::SerializationFailed("time".to_string()))?;
+            map.insert("time", formatted);
```

Comment:
Is this `format` ever able to fail? Otherwise I think we should just replace this with a `expect`

Reply:
looking at the source code of `time` its theoretically possible to fail if their `write` calls to the target string fail (maybe because of oom?) but that seems very unrealistic so ill change it to a `except`